src/main/java/marquez/client/models/Dataset.java

```diff
@@ -29,11 +32,22 @@
 @AllArgsConstructor(onConstructor = @__(@JsonCreator))
 @EqualsAndHashCode
 @ToString
-public final class Dataset {
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+@JsonSubTypes({
+  @JsonSubTypes.Type(value = DbTable.class, name = "DB_TABLE"),
+  @JsonSubTypes.Type(value = Stream.class, name = "STREAM")
+})
+public abstract class Dataset {
+  @Getter
+  @NonNull
+  @JsonProperty("type")
```

Comment:
_Minor:_ We can remove the use of `@JsonProperty` given that we set the `-parameter` jvm flag.

Reply:
Will do!